 <launch>

 
  <arg name="autonomous_mode" default="false" />
  <arg name="launch_battery_monitor" default="false" />
  <arg name="wheel_namespace" default="wheels" />
  <arg name="wheel_config_type" default="differential_drive" />
  <arg name="speed_controller_loop_rate" default="20.0" />

  <param name="wheel_namespace" type="str" value="$(arg wheel_namespace)" />
  <param name="wheel_config_type" type="str" value="$(arg wheel_config_type)" />
  <param name="speed_controller_loop_rate" type="double" value="$(arg speed_controller_loop_rate)" />
  <param name="speed_controller_use_pid" type="bool" value="false" />

  <env name="ROSCONSOLE_CONFIG_FILE"
        value="$(find toulouse_rover)/param/rosconsole.config"/>

  <node pkg="toulouse_rover" name="serial_motor_driver" type="serial_motor_driver.py" output="screen">  </node>

  <node pkg="rosserial_python" name="serial_node" type="serial_node.py" if="$(arg launch_battery_monitor)" output="screen" args="_port:=/dev/ttyUSB0 _baud:=57600">
  </node>
  
  <node pkg="toulouse_rover" name="wheel_speed_calculator_node" type="wheel_speed_calculator_node"  output="screen"> </node>

  <node pkg="toulouse_rover" name="wheel_speed_controller_node" type="wheel_speed_controller_node"  output="screen">
    <rosparam command="load" file="$(find toulouse_rover)/param/wheel_speed_controller_params.yaml" />
  </node>

  <!-- -->
  <node pkg="toulouse_rover" name="odom_calculator_node" type="odom_calculator_node"  output="screen">
    <param name="odom_topic" value="odom_wheels" />
    <param name="odom_frame_id" value="odom_wheels" />
  </node>

  <!-- loading urdf model of robot -->
  <param name="robot_description" textfile="$(find toulouse_rover)/urdf/toulouse.urdf.xacro"/>
  
  <node pkg="joint_state_publisher" type="joint_state_publisher" name="joint_state_publisher" />
  <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" />


  <include file="$(find toulouse_rover)/launch/toulouse_autonomy.launch" if="$(arg autonomous_mode)">
  </include>

</launch>